### PR TITLE
Let's compute the root_fields_to_add_to_expanded_event only once per …

### PR DESCRIPTION
**Pull request by aspacca**
…events to expand

Enhancement
## What does this PR do?
Let's compute the `root_fields_to_add_to_expanded_event` only once per events to expand in order to improve performance

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
~- [ ] I have made corresponding change to the default configuration files~
~- [ ] I have added tests that prove my fix is effective or that my feature works~
- [x] I have added an entry in `CHANGELOG.md`

## Author's Checklist

- [ ]

## Related issues

-

### CODE REVIEW

File: docs/en/aws-elastic-serverless-forwarder-configuration.asciidoc [hunk 1]
```diff
@@ -288,6 +290,13 @@ inputs:
 
 You can define `inputs.[].expand_event_list_from_field` as a string with the value of a key in the JSON that contains a list of elements that must be sent as events instead of the encompassing JSON.
 
+If you want to inject in the expanded events, fields from the root of the JSON object that you are expanding from, you can define `inputs.[].root_fields_to_add_to_expanded_event`. The config takes two different type of value:
```
Comment (dedemorton) [inline docs/en/aws-elastic-serverless-forwarder-configuration.asciidoc:293]: This description is hard to follow. Would you mind adjusting the example in this topic to include the `root_fields_to_add_to_expanded_event` setting so that I (and users) can better understand what you are trying to say here? Then I can offer edits.

I think the big question I need answered is: What do we mean by root fields? I'm assuming we mean the fields under `Records`, but seeing a realistic example would help me understand. Thanks!

Comment (aspacca) [reply docs/en/aws-elastic-serverless-forwarder-configuration.asciidoc:293]: @dedemorton the example is present: https://github.com/elastic/elastic-serverless-forwarder/pull/642/files/5d1d09b9a65bafee56ce49a4dccd434e9ed96c98#diff-63801152050878371d35100604aa7d8d328f9d97bdecba179bb617554bbae968R280-R281 :)


The root fields are the fields at the root of the JSON object that's expanded, so at the same level for `Records`, for example:
```
{ "Records": [...], "owner": 1234567890, "logGroup": "a log group", "logStream": "a log stream" }
```

File: share/expand_event_list_from_field.py [hunk 2]
```diff
@@ -42,29 +42,29 @@ def _expand_event_list_from_field(
...
+            # we want to add only a list of root fields
+            elif isinstance(self._root_fields_to_add_to_expanded_event, list):
+                for root_field_to_add_to_expanded_event in self._root_fields_to_add_to_expanded_event:
+                    if root_field_to_add_to_expanded_event in json_object:
+                        root_fields_to_add_to_expanded_event[root_field_to_add_to_expanded_event] = json_object[
+                            root_field_to_add_to_expanded_event
+                        ]
+                    else:
+                        shared_logger.debug(
+                            f"`{root_field_to_add_to_expanded_event}` field to be added"
```
Comment (kaiyan-sheng) [inline share/expand_event_list_from_field.py:59]:
```suggestion
                            f"`{root_field_to_add_to_expanded_event}` field specified in `root_fields_to_add_to_expanded_event` parameter is not present at root level"
```

**Review by kaiyan-sheng — DISMISSED** [review]
One minor comment on the error message, everything looks good to me.

File: docs/en/aws-elastic-serverless-forwarder-configuration.asciidoc (hunk 1 above)
Comment (dedemorton) [reply docs/en/aws-elastic-serverless-forwarder-configuration.asciidoc:293]: @aspacca I didn't mean the configuration example. I meant it would help to see how the `root_fields_to_add_to_expanded_event` setting affects the data that is shown under the [Example section](https://www.elastic.co/guide/en/esf/current/aws-elastic-serverless-forwarder-configuration.html#_example) in the docs (where it shows how the setting affects the events that are produced).

**Review by kaiyan-sheng — APPROVED** [review]

File: docs/en/aws-elastic-serverless-forwarder-configuration.asciidoc [hunk 3]
```diff
@@ -288,10 +290,17 @@ inputs:
...
```
Comment (dedemorton) [inline docs/en/aws-elastic-serverless-forwarder-configuration.asciidoc:293]: I think removing some of the words here will make the sentence easier to parse without losing meaning (I hope):
```suggestion
To inject root fields from the JSON object into the expanded events, define `inputs.[].root_fields_to_add_to_expanded_event`. The config takes one of the following values:
```

File: docs/en/aws-elastic-serverless-forwarder-configuration.asciidoc [hunk 4]
```diff
@@ -288,10 +290,17 @@ inputs:
...
+
+. the literal string "**all**", in order to inject all the fields at root level (but for the one you are expanding events from): `root_fields_to_add_to_expanded_event: "all"`
```
Comment (dedemorton) [inline docs/en/aws-elastic-serverless-forwarder-configuration.asciidoc:295]: I think removing some unnecessary words here makes the intention a little clearer.
```suggestion
. the literal string "**all**" to inject all root fields (except the field you are expanding events from). For example, `root_fields_to_add_to_expanded_event: "all"`
```

File: docs/en/aws-elastic-serverless-forwarder-configuration.asciidoc [hunk 5]
```diff
@@ -288,10 +290,17 @@ inputs:
...
+. a list of the names of the fields at root level that you want to inject: for example  `root_fields_to_add_to_expanded_event: ["owner", "logGroup", "logStream"]`
```
Comment (dedemorton) [inline docs/en/aws-elastic-serverless-forwarder-configuration.asciidoc:296]: Gain I think that too many words actually make this sentence harder to parse. How about:
```suggestion
. a list of the root fields that you want to inject. For example,  `root_fields_to_add_to_expanded_event: ["owner", "logGroup", "logStream"]` 
```

**Review by dedemorton — COMMENTED** [review]
Thanks for adding the example. I think it makes the intention of these configs clearer.

I've edited the descriptions to make them more straightforward. Hopefully I have not inadvertently changed the meaning.

**Review by dedemorton — APPROVED** [review]
LGTM
